GVFS/GVFS/CommandLine/GVFSVerb.cs

```diff
@@ -113,7 +113,7 @@ public static bool TrySetRequiredGitConfigSettings(Enlistment enlistment)
                 { "core.safecrlf", "false" },
                 { "core.untrackedCache", "false" },
                 { "core.repositoryformatversion", "0" },
-                { "core.filemode", "false" },
+                { "core.filemode", GVFSPlatform.Instance.GitCoreFileMode ? "true" : "false" },

```

Comment:
Do we need to explicitly set this here? I think this will normally be set appropriately by the platform specific git that is running, so on Windows, this is set to false, and on Mac, this is set to True. Does GVFS need to know about and explicitly set this?

Reply:
> Does GVFS need to know about and explicitly set this?

My recollection is that we set this explicitly in case there is a global or system git config that sets `core.filemode` to something other than the platform default.

Reply:
Looking at history and this was added when the repair was added for the .git/config so when it is corrupted we can fix it by putting the right values in.

Reply:
> Looking at history and this was added when the repair was added for the .git/config so when it is corrupted we can fix it by putting the right values in.

Yep, for future reference, this was the conversation on this PR that led to this change (note the initial changes in the PR were to call `git init` when repairing the config):

>Saeed 7/10/2017
Why are we calling 'git init'? Why not just create a new empty config file?
It looks like that's what you're doing down below. I don't understand what the 'git init' is for.


> Forest 7/10/2017
git init populates four entries that GVFS does not. I don't know much about them, but I figured this is how clone initializes the config, so it was best to keep it in line with that.

>    [core]
repositoryformatversion = 0
filemode = false
bare = false
logallrefupdates = true


>Saeed 7/11/2017
It was just surprising to see "git init" called on an existing repo. What other side effects does it have besides recreating the config file?
If it's just a matter of creating those 4 config entries, I'd be fine with adding them to the list of entries in GVFSVerb, since they all seem like sensible values, and we're currently relying on them anyway so might as well make it explicit.


>Forest 7/11/2017
From the git man:
Running git init in an existing repository is safe. It will not overwrite things that are already there. The primary reason for rerunning git init is to pick up newly added templates (or to move the repository to another place if --separate-git-dir is given).
> So really, the primary side effect would be "accidentally fixing" any missing folders in .git (refs/heads, objects, etc). but I'm not opposed to just adding those entries to simplify/clarify things here.